cloudinit/config/schema.py

```diff
@@ -797,8 +797,9 @@ def validate_cloudconfig_schema(
         ):  # pylint: disable=W1116
             if (
                 "devel" != features.DEPRECATION_INFO_BOUNDARY
-                and Version.from_str(schema_error.version)
-                > Version.from_str(features.DEPRECATION_INFO_BOUNDARY)
+                and (schema_error.version == "devel"
+                or Version.from_str(schema_error.version)
+                > Version.from_str(features.DEPRECATION_INFO_BOUNDARY))
```

Comment:
this won't work for our default "devel" case as Version.from_str("devel") will traceback
```
ValueError: invalid lteral for int() with base 10: 'devel'
```

Reply:
Resolving this comment as we only see schema_error.version == "devel" when no deprecated_version is set in the schema definition, which happens to be when we have the changed_version changed_message keys... Which I realize is what @holmanb was proposing to fix here.

I also can confirm this fix doesn't report deprecation messages for "changed" keys